client/lib/url-search/use-url-search.ts

```diff
@@ -0,0 +1,79 @@
+import { useState } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+import debugFactory from 'debug';
+import page from 'page';
+
+const debug = debugFactory( 'calypso:url-search' );
+
+/**
+ * Function for constructing the url to page to. Here are some examples:
+ * 1. { uri: 'google.com', search:'hello' } --> 'google.com?s=hello'
+ * 2. {
+ *     uri: 'wordpress.com/read/search?q=reader+is+awesome',
+ *     search: 'reader is super awesome'
+ *     queryKey: 'q',
+ *    } --> 'wordpress.com/read/search?q=reader+is+super+awesome'
+ *
+ * @param {object} options the options object
+ * @param {string} options.uri the base uri to modify and add a query to
+ * @param {string} options.search the search term
+ * @param {string} [options.queryKey = s] the key to place in the url.  defaults to s
+ * @returns {string} The built search url
+ */
+export const buildSearchUrl = ( {
```

Comment:
Any reason why this function isn't designed to be a hook? Looks like it will be a good idea, especially considering that we're already exporting it (even if only for testing purposes at this time).

Reply:
It's [based on this file](https://github.com/Automattic/wp-calypso/blob/replace_urlsearch_wrapper/client/lib/url-search/index.js#L25), so we could make it as a hook, but if I follow your suggestion, I receive this error:

<img width="857" alt="image" src="https://user-images.githubusercontent.com/1044309/167212015-dd87614c-e1e7-4876-9113-6e6739b62f5b.png">

`React Hook "useBuildSearchUrl" is called in function "doSearch" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use"`

Do you have a suggestion about how to fix it?

Reply:
I just wouldn't export this function. Test the hook instead.

Reply:
Thanks for the suggestion @lsl, I changed it here: https://github.com/Automattic/wp-calypso/pull/63352/commits/acb1c40cbd8801683c303acb2a637a1e924795a8